azkaban-web-server/src/main/java/azkaban/webapp/servlet/ProjectManagerServlet.java

```diff
@@ -688,17 +688,22 @@ private void ajaxFetchJobInfo(final Project project, final HashMap<String, Objec
       return;
     }
 
-    final Props prop;
+    Props prop;
     try {
-      prop = this.projectManager.getProperties(project, node.getJobSource());
+      prop = this.projectManager.getProperties(project, flow, jobName, node.getJobSource());
     } catch (final ProjectManagerException e) {
       ret.put("error", "Failed to retrieve job properties!");
       return;
     }
 
+    if (prop == null) {
```

Comment:
Why necessary to new here?

Reply:
Because if prop is null, below method `prop.getKeySet()` at line 716 would throw null pointer exception.